### PR TITLE
Add logic to connect API

### PR DESCRIPTION
**Pull request by Rayan84**
In this branch I added the following features:

- Generate API unique code
- Logic to add new entries to the API
- Logic to remove entries from the API

### CODE REVIEW

File: src/redux/books/books.js [hunk 1]
```diff
@@ -1,35 +1,97 @@
...
+    .delete(`${URL}/${payload.id}`)
+    .then((response) => {
+      if (response.data === 'The book was deleted successfully!') {
+        dispatch(fetchBooks());
+      }
+    })
+    .catch((error) => {
+      fetchBooksFailure(error.message);
+    });
+};
```
Comment (omar-labana) [inline src/redux/books/books.js:74]: The redux code of the application should be refactored using *middleware* to make `async` requests to the Bookstore API. Kindly, update your functions to make these requests asynchronously(using `async` and `await`).

**Review by omar-labana — CHANGES_REQUESTED** [review]
Hi @Rayan84, 

Excellent work so far! 👌
You have implemented almost all the required features. ✔️ 
There are some cases that you need to adjust but you are almost done! 😀

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/redux/books/books.js (hunk 1 above)
Comment (Rayan84) [reply src/redux/books/books.js:74]: I am using axios .then instead of async await

**Comment by omar-labana** [comment]
Hi @Rayan84, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**